core/src/main/java/org/infinispan/stream/impl/LocalStreamManagerImpl.java

```diff
@@ -129,9 +132,14 @@ public void inject(Cache<K, V> cache) {
    public void start() {
       cacheName = ByteString.fromString(cache.getName());
       hasLoader = cache.getCacheConfiguration().persistence().usingStores();
-      this.maxSegment = cache.getCacheConfiguration().clustering().hash().numSegments();
+      ClusteringConfiguration clusteringConfiguration = cache.getCacheConfiguration().clustering();
+      this.maxSegment = clusteringConfiguration.hash().numSegments();
       localAddress = rpc.getAddress();
-      cache.addListener(this);
+      // Replicated never removes segments so we don't have to worry about rehash event
+      requiresListener = !clusteringConfiguration.cacheMode().isReplicated();
```

Comment:
I think using a `isReplicated` field might actually be clearer, since in some cases we still use a listener.

Reply:
Sure, sounds good.